pkg/plugins/golang/deploy-image/v1alpha1/scaffolds/internal/templates/api/types.go

```diff
@@ -106,6 +105,9 @@ type {{ .Resource.Kind }}Status struct {
 	// The status of each condition is one of True, False, or Unknown.
 	// +listType=map
 	// +listMapKey=type
+	// +patchStrategy=merge
+	// +patchMergeKey=type
+	// +optional
```

Comment:
@JoelSpeed without the markers we will see 

```
conditions: Conditions field is missing the following markers: patchStrategy=merge, patchMergeKey=type (kubeapilinter)
	Conditions []metav1.Condition `json:"conditions,omitempty"`
```

It seems right, without those, I understand that we can end up replacing the whole list when just one condition is updated.

Reply:
It is a pity that sig-api-machinery and sig-cli cannot coordinate these things. Forcing users to "duplicate" these markers on any array field is just sad. They have (almost) the same semantics.

Reply:
`patchStrategy` and `patchMergeKey` only work for built-ins, they do not work for CRDs. There is configuration in the linter for conditions to tell it to ignore their absence.

Reply:
I will remove but would be nice if we could to do so

Reply:
It's on the todo list, I'd like to have an option to default the configuration to "CRD" instead of built-in, but I haven't explored this yet